lib/ama_layout/notification.rb

```diff
@@ -0,0 +1,65 @@
+module AmaLayout
+  class Notification
+    TYPES = %i[notice warning alert].freeze
+    FORMAT_VERSION = '1.0.0'.freeze
+
+    # NOTE: The following attributes are designed to be immutable - you need
+    # make a new instance to change them. The only mutable attribute is :active.
+    attr_reader :id, :type, :header, :content, :created_at, :version
+    attr_accessor :active
+
+    def initialize(args = {})
+      args = args.with_indifferent_access
+      @id = args[:id]
+      @type = args.fetch(:type, :notice).to_sym
+      @header = args.fetch(:header)
+      @content = args.fetch(:content)
+      @created_at = parse_time(args.fetch(:created_at))
+      @version = args.fetch(:version, FORMAT_VERSION)
+      self.active = args.fetch(:active)
+      invalid_type! if TYPES.exclude?(type)
```

Comment:
Personally, I'd prefer to fail immediately in this case. We wouldn't ever want non-specified notification types.

If this was a validation, we'd have to keep calling `valid?` as well as including `ActiveModel::Model` or something similar.

Breaking execution with an exception gives us a guarantee on the data values in the data store.